python/setup.py

```diff
@@ -171,6 +171,17 @@ def get_packages(self):
 # in this directory
 if setup_spec.type == SetupType.RAY:
     setup_spec.extras = {
+        "data": [
+            "dask",
+            "mars",
+            "modin",
+            "pyspark",
+            "tensorflow",
+            "torch",
```

Comment:
This is excessive, can you please remove these first 6? We should add runtime import checks.